### ISSUE TITLE
Brain mask nii derivative file filled with NaN values

### ISSUE THREAD

**Issue by adimaron** [nipreps/fmriprep#2503]
### What version of fMRIPrep are you using?
**20.2.1**
### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?
**Docker**
### What is the exact command-line you used?
fmriprep-docker \
	  	--fs-license-file "/usr/local/freesurfer/license.txt" \
		"${input_dir_fmriprep}" \
		"${preprocess_dir}" \
		participant \
		--aroma-melodic-dimensionality=-200 \
		--bold2t1w-dof=6 \
		--bold2t1w-init=register \
		--dvars-spike-threshold=1.5 \
		--fd-spike-threshold=0.5 \
		--fs-no-reconall\
		--ignore fieldmaps slicetiming \
		--mem_mb="${mem}" \
		--n_cpus="${cpus}" \
		--output-spaces MNI152NLin2009cAsym anat \
		--skull-strip-t1w=force \
		--skull-strip-template=OASIS30ANTs \
		--use-syn-sdc \
		--verbose \
		--notrack \
		--participant-label "${subject}" \
		--bids-filter-file "${working_dir}/filter.json" \
		-w "${output_dir_fmriprep}/fmriprep_working"

### Have you checked that your inputs are BIDS valid?
Yes
### Did fMRIPrep generate the visual report for this particular subject? If yes, could you share it?
Yes it did.
Attached as pdf

### Can you find some traces of the error reported in the visual report (at the bottom) or in *crashfiles*?
No
<please copy&paste all the information you can gather about errors>
```

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?
No

### fMRIPrep log
If you have access to the output logged by fMRIPrep, please make sure to attach it as a text file to this issue.
Attached
[fmriPrepLog_B001_fromYann.txt](https://github.com/nipreps/fmriprep/files/6972205/fmriPrepLog_B001_fromYann.txt)

<!--
List the steps you performed that revealed the bug to you.

I found that downstream steps that require the derivative file ending with brain_mask.nii.gz are failing, and then opened it in an nii viewer and found that it contained only NaN values. I found this across multiple runs with this machine and this Docker, but not on another machine that also uses fmriprep-docker.
I am not sure if this qualifies as a bug but cannot find any reason to why it is happening
The functional data output , sub-B001_ses-baseline_task-resting_space-T1w_desc-preproc_bold.nii.gz looks good.

Include any code samples. Enclose them in triple back-ticks (```)
Like this:

```
<code>
```
-->
[fmriPrepLog_B001_fromYann.txt](https://github.com/nipreps/fmriprep/files/6972212/fmriPrepLog_B001_fromYann.txt)
[B001_visualReport.pdf](https://github.com/nipreps/fmriprep/files/6972214/B001_visualReport.pdf)

**Comment by lorisnaspi** [comment]
Hi Adimaron,

I have the same problem. Brain masks derivatives are empty. I cannot even open them with MRICron. My final report does not indicate any errors. Have you solved the problem

**Comment by lorisnaspi** [comment]
In my processing stream I have noticed a WARNINGS where the brain_mask.nii.gz files are transformed from float32 to uint8. This is the only warning that I get for those .nii images (only brain masks) that I cannot open. I do not know if this is a bug or what

**Comment by julfou81** [comment]
Hi,

We noticed this also on our processed brain_mask.nii.gz images with fmriprep v20.2.2. We are able to display them with fsleyes but this presence of NaN instead of zeros makes SPM crash.
However when we preprocessed the same images with fmriprep v20.2.1, we did not noticed the same behavior, we had no issue to use SPM on them.

For now, we add this line in our script as a small patch on this issue:
```
fslmaths in_image -nan out_image

 -nan   : replace NaNs (improper numbers) with 0
 -nanm  : make NaN (improper number) mask with 1 for NaN voxels, 0 otherwise
```
